### PR TITLE
Spatial texture

### PR DESCRIPTION
**Pull request by hollasch**

### CODE REVIEW

**Comment by trevordblack** [comment]
Add update to CHANGELOG

File: books/RayTracingTheNextWeek.html [hunk 1]
```diff
@@ -1040,11 +1040,32 @@
...
+
+To begin with, we'll make the texture colors procedural, and will create a texture map of constant
+color. Most programs keep constant RGB colors and textures in different classes, so feel free to do
+something different, but I am a big believer in this architecture because it's great being able to
+make any color a texture.
+
+In order to perform the texture lookup, we need a _texture coordinate_. This coordinate can be
+defined in many ways, and we'll develop this idea as we progress. For now, we'll pass in two
+dimensional texture coordinates. By convention, such coordinates are $u$ and $v$. For a constant
+texture, every $u,v$ pair yields a constant color, so we can actually ignore the coordinates
```
Comment (trevordblack) [inline books/RayTracingTheNextWeek.html:1061]: $(u,v)$ ?

Not sure myself.

File: books/RayTracingTheNextWeek.html [hunk 2]
```diff
@@ -1097,47 +1118,64 @@
...
+`hittable`.
 
 
 Solid Textures: A Checker Texture
 ----------------------------------
-<div class='together'>
-We can create a checker texture by noting that the sign of sine and cosine just alternates in a
-regular way, and if we multiply trig functions in all three dimensions, the sign of that product
-forms a 3D checker pattern.
+The next simplest texture map type is the solid texture, which depends only on the position of each
```
Comment (trevordblack) [inline books/RayTracingTheNextWeek.html:1129]: "The simplest texture map type is the solid texture..." ?

File: books/RayTracingTheNextWeek.html [hunk 3]
```diff
@@ -1097,47 +1118,64 @@
...
+point in 3D space. You can think of a solid texture (of color) as coloring space itself, visible on
+the object surface. For this reason, the object can move through the colors of the texture as it
```
Comment (trevordblack) [inline books/RayTracingTheNextWeek.html:1131]: "as coloring space itself" Do you mean that
1. The solid texture is coloring (verb) the space
2. A solid texture is a kind of colored (adjective) space

This sentence took me a moment to parse. I'm still not actually sure what you mean

File: books/RayTracingTheNextWeek.html [hunk 4]
```diff
@@ -1097,47 +1118,64 @@
...
+changes position, though usually you would want to fix the relationship between the object and the
+solid texture.
+
+To accomplish the checkered pattern, we'll first compute the floor of each component of the input
+point. We could truncate the coordinates, but that would pull values toward zero, which would give
```
Comment (trevordblack) [inline books/RayTracingTheNextWeek.html:1136]: you might want to reinforce that the input point is $(u,v)$

File: books/RayTracingTheNextWeek.html [hunk 5]
```diff
@@ -1097,47 +1118,64 @@
...
+us the same color on both sides of zero. The floor function will always shift values to the integer
+value on the left (toward negative infinity). Given these three integer results, we take their sum
```
Comment (trevordblack) [inline books/RayTracingTheNextWeek.html:1138]: Maybe:

"Given these three integer results, $floor(x), floor(y), floor(z)$, we take their ...

File: books/RayTracingTheNextWeek.html [hunk 6]
```diff
@@ -1222,13 +1260,22 @@
...
+want to get a consistent effect on the surface of our objects. That approach is covered next.
 
 
 Texture Coordinates for Spheres
 --------------------------------
+Constant-color textures use no coordinates. Solid (or spatial) textures use the coordinates of a
+point in space. Now it's time to make use of the $u,v$ texture coordinates. These coordinates
+specify the location on 2D source image (or in some 2D parameterized space). To get this, we need a
+way to find the $u,v$ coordinates of any point on the surface of a 3D object. This mapping is
+completely arbitrary, but generally you'd like to have cover the entire surface, and scale, orient
```
Comment (trevordblack) [inline books/RayTracingTheNextWeek.html:1275]: "... you'd like to cover ..."

Comment (trevordblack) [reply books/RayTracingTheNextWeek.html:1275]: "completely arbitrary, but generally you'd like to cover the entire surface, and be able to scale, orient, and stretch ..."

Feel free to make a judgement call on the oxford comma

**Comment by trevordblack** [comment]
Sorry for the small little nitpicks.
Otherwise, this is fantastic work

**Review by trevordblack — APPROVED** [review]

File: books/RayTracingTheNextWeek.html (hunk 1 above)
Comment (hollasch) [reply books/RayTracingTheNextWeek.html:1061]: Updated.

File: books/RayTracingTheNextWeek.html (hunk 2 above)
Comment (hollasch) [reply books/RayTracingTheNextWeek.html:1129]: I was asserting that if the solid color is the simplest texture map, the spatial/solid texture is the next simplest. So technically correct, but it struck me as awkward too. Rethinking...

File: books/RayTracingTheNextWeek.html (hunk 3 above)
Comment (hollasch) [reply books/RayTracingTheNextWeek.html:1131]: Reworded.

File: books/RayTracingTheNextWeek.html (hunk 5 above)
Comment (hollasch) [reply books/RayTracingTheNextWeek.html:1138]: Done.

File: books/RayTracingTheNextWeek.html (hunk 4 above)
Comment (hollasch) [reply books/RayTracingTheNextWeek.html:1136]: Yeah, that triggered me to write an entire paragraph that was missing.

**Comment by hollasch** [comment]
Thank you! That was great feedback, and definitely helped.

**Review by trevordblack — APPROVED** [review]
Oh! Good changes

**Pull request merged**